batch/testResultsCollector.js

```diff
@@ -1,40 +1,52 @@
 import {XUnitParser} from './parsers/xUnitParser';
-import mongoose from 'mongoose';
-import {TestSuiteModel} from '../models/testSuiteModel';
-import {TestCaseModel} from '../models/testCaseModel';
-
+import {GithubHelper} from '../helpers/githubHelper';
+import LudwigDAO from '../database/ludwigDAO';
 
 class TestResultsCollector {
-	constructor(configuration) {
-		this.configuration = configuration;
-		mongoose.connect(configuration.mongo.uri, configuration.mongo.options);
+	constructor(ludwigConfiguration) {
+		this.configuration = ludwigConfiguration;
+		this._xUnitParser = new XUnitParser(this.configuration);
+		this._githubHelper = new GithubHelper(this.configuration);
+		this._dao = LudwigDAO;
+	}
+
+	get parser() {
+		return this._xUnitParser;
+	}
+
+	get githubHelper() {
+		return this._githubHelper;
```

Comment:
Same as above, the getters are there because I haven't found a simpler way to stub what I need for my tests (and I need some kind of access from the outside to be able to do that).